andela/ninjas-ah-backend
# #166326326 favorite an article

#### What does this PR do?
Add endpoints to allow a user to favorite an article
#### Description of Task to be completed?
Have the following endpoints working and accessible to an authenticated user
- `PATCH /api/v1/articles/:slug/favorite`
- `DELETE /api/v1/articles/:slug/favorite`
- `GET /api/v1/articles/favorited`

#### How should this be manually tested?
- After cloning the repo, cd into it and type `npm test`
- Using Postman, firstly make sure you signup then login to get the access token, secondly make sure you pass the access token in the header with the following key name: `access-token`, secondly create an article, and finally make any request to the above endpoints
- Using the mockup,  after running the server by typing `npm run start` or `npm run dev`, firstly create an account by visiting `http://localhost:<PORT>/mockups/signup.html`, then login by visiting `http://localhost:<PORT>/mockups/login.html` to get an access token, after that create an article by visiting  `http://localhost:<PORT>/mockups/articles.html`, finally visit `http://localhost:<PORT>/mockups/favorite_article.html` to start performing any request

#### What are the relevant pivotal tracker stories?
[166326326](https://www.pivotaltracker.com/story/show/166326326)

#### Screenshots
![image](https://user-images.githubusercontent.com/44000547/58618250-6a49d280-82c2-11e9-9769-91ba8787a443.png)
